src/main/java/core/basesyntax/service/impl/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.WarehouseDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final Map<String, Integer> warehouse = new WarehouseDaoImpl().getWarehouse();
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        int currentFruitQuantity = warehouse.get(transaction.getFruit());
+        int newFruitQuantity = currentFruitQuantity - transaction.getQuantity();
+        String fruit = transaction.getFruit();
+        if (currentFruitQuantity >= transaction.getQuantity()) {
+            warehouse.put(fruit, newFruitQuantity);
```

Comment:
do not operate with the warehouse directly. Instead, use DAO to update the warehouse state. fix in all handlers